### ISSUE TITLE
Copy with just content="stuff" always returns changed even if the content does not change

### ISSUE THREAD

**Issue by vrischmann** [ansible/ansible#3947]
Hi,

I had a task like this:

```
- name: Set /etc/timezone
  copy: content="Europe/Paris" dest=/etc/timezone
  register: _timezone

- name: debug
  debug: msg="{{ _timezone }}"
```

The problem is it always changes, even if the value is identical. Looking at the source code from the copy module, it always does a md5sum on the src argument, so it can never work.

**Comment by bcoca** [comment]
It might be fixable by changing how content works, so it now creates a tmp
file that then gets passed to src, then copy can proceed as usual.

**Comment by jimi-c** [comment]
I am unable to reproduce this with the current development branch. What version of ansible are you testing against?

**Comment by vrischmann** [comment]
I just tested using the latest master from Git.
The single test case I posted works if I put it alone in a test site.yml file like this:

```
---
- hosts: all
  user: root
  tasks:
    - name: Set /etc/timezone
      copy: content="Europe/Paris" dest=/etc/timezone
      register: _timezone

    - name: debug
      debug: msg="{{ _timezone }}"
```

Which I then use with `ansible-playbook -i hosts site.yml`, and it works.

However if I put the test case in a role in the project where I discovered this problem, it still does not work.
I'll try to debug.

**Comment by jimi-c** [comment]
Yes, I was testing with what you posted above and saw the same result - things worked as expected. Can you share the playbook/tasks as configure when it does not work?

**Comment by vrischmann** [comment]
Something I didn't mention before, but I install Ansible in my virtualenv via pip with `git+https://github.com/ansible/ansible.git` in my requirements.txt file. If I'm not wrong it installs the last development version ?

So I just tested again with my playbook, this time creating a new virtualenv from scratch and installing Ansible, and it looks like it works now, maybe I screwed something up before and still used the old virtualenv, I'm not sure.

Also something I did'nt mention, when it didn't work I was using the same procedure to install Ansible as I said earlier, and I did that installation less than two weeks ago, so I guess something must have changed in Ansible which makes it work now.

Anyway, I guess it's fixed.

**Comment by jimi-c** [comment]
When installing via pip, it is the official release version (1.2.3 at this point). To use devel, you must checkout the source code via git and either install it or source the hacking/env-setup script.

I'll go ahead and close this, as it does appear fixed and 1.3 is scheduled for release this Friday. Thanks!
